nuxeo-features/nuxeo-platform-publisher/nuxeo-platform-publisher-core/src/main/java/org/nuxeo/ecm/platform/publisher/impl/service/PublisherServiceImpl.java

```diff
@@ -99,7 +99,9 @@
 
     @Override
     public void applicationStarted(ComponentContext context) {
-        if (TransactionHelper.startTransaction()) {
+        boolean started = TransactionHelper.isTransactionActive();
+
+        if (started || TransactionHelper.startTransaction()) {
```

Comment:
what about the following statement instead?
`if (started = TransactionHelper.isTransactionActive() || TransactionHelper.startTransaction())`

See the following remark about the call to `TransactionHelper.commitOrRollbackTransaction()` if `started`